### PR TITLE
useMediaQuery to check window height to fix welcome modal vertical re…

### LINKED ISSUE #6731: [Frontend] Welcome modal doesn't resize vertically with screen
**Describe the bug**
The Welcome modal that is displayed when entering the website for the first time does not scale vertically based on the screen size or allow scrolling down to see the whole modal.

**To Reproduce**
1. Open a private browser (or clear history for https://app.electricitymaps.com/map)
2. Go to https://app.electricitymaps.com/map
3. Make browser window smaller vertically

**Expected behavior**
I expected the Welcome modal to either resize or the page to allow me to scroll so that the rest of the modal would come into view.

**Screenshots**
image

**Desktop (if relevant):**
 - OS: Windows 10
 - Browser: Firefox and Chrome

### PR DESCRIPTION
**Pull request by noodlebenji2960**
…size #6731

## Issue 6731
Closes #6731

## Description
The Welcome modal on the website does not adjust its vertical size based on the screen size, as per the issues thread it was preferred to be shrunken to fit the screen size rather than scrolling.

Fix:
To address this issue, I utilized the existing useMediaQuery hook to dynamically check the height of the viewport. Based on this height measurement, conditional styles were applied to accomodate the smaller size modal i.e. flex-col.

This is my first pull request ever. please be nice.

### Preview
issue6731-2024-07-03

- [✔] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [✔] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

**Review by VIKTORVAV99 — COMMENTED** [review]
Thanks for taking the time to open your very first PR here in the Electricity Maps contrib repository!

The changes you have made so far looks good and there are no changes needed in the code you have already altered!

I would however ask you to take a look and see if you can do some minor tweaks to the styling so things look bit more uniform. These are purely optional and if you don't feel comfortable doing them please let me know and we can merge this PR without them :).

It would be great if the text could be centred here (and logo if you think that makes sense):
image

It would also be great if these corners could be rounded to match the ones on the left side (strangely enough it looks correct on the first one):
image

I also noticed there are some alignment issues with the images but I think we might need take some new images with uniform aspect rations and sizes to fix that. So I think we can leave those as is.

Please don't hesitate to reach out and ping me if you have any questions.

PS: I pushed a commit that fixes the prettier formatting as there was a conflicting ESLint rule that would change things back. I'll fix that in another PR when able.

**Comment by noodlebenji2960** [comment]
_"It would be great if the text could be centred here (and logo if you think that makes sense)"_
Yipee, I'm glad it is up to standard. I will get on this now.

_"It would also be great if these corners could be rounded to match the ones on the left side (strangely enough it looks correct on the first one)"_
This weirdly seems to be an issue only in dark mode from what I briefly looked at. I will investigate further and see if I can fix it.

"I also noticed there are some alignment issues with the images but I think we might need take some new images with uniform aspect rations and sizes to fix that. So I think we can leave those as is."
Yeah this was a little challenging to work around but I did what I could while only making minimal/essential changes outlined respective to the outlined issue. Uniform aspect ratios will definitely fix this.

Thanks for the support - Ben

**Comment by noodlebenji2960** [comment]
Okay i think i've made the relevant changes and I hope I have commited them properly:

- Removed the "dark:bg-gray-700" from the currentview.rendercontent(t) div container which fixed the sharp corners issue in dark mode.
- Added a conditional max width to the logo for the first view which seems to have fixed the text alignment.
-Added some conditonal padding so text isn't overlapped by the "X" close-modal button when window is small vertically and horizonatally.

Also just to further reinforce your point that new images will be required, the image for "switch between electricity and emissions views" has some rounded corners as part of the png file, they should probably be rounded using tailwind instead.  Additonally, on one of the views some of the text escapes the modal when the window is very small both vertically and horizontally. This will probably be more easily addressed when the new images are being implemented.

File: web/src/components/modals/OnboardingModalInner.tsx [hunk 1]
```diff
@@ -94,10 +106,16 @@ function Modal({
...
+            className={`flex w-full grow self-center
+              ${
+                isMinHeight ? 'h-1/2 max-h-[264px] rounded-t-3xl' : 'h-full rounded-l-3xl'
+              } bg-auto bg-center bg-no-repeat ${
+              isOnFirstView()
+                ? `dark:invert ${isMinHeight ? 'max-w-[10rem]' : 'max-w-[400px]'}`
+                : isMinHeight
+                ? ''
+                : 'max-w-[400px]'
+            }`}
```
Comment (VIKTORVAV99) [inline web/src/components/modals/OnboardingModalInner.tsx:118]:
```suggestion
            className={`flex w-full grow self-center
              ${
                isMinHeight ? 'h-1/2 max-h-[264px] rounded-t-3xl' : 'h-full rounded-l-3xl'
              } bg-auto bg-center bg-no-repeat ${
              isOnFirstView()
                ? `dark:invert ${isMinHeight ? 'max-w-[10rem]' : 'max-w-[400px]'}`
                : !isMinHeight && 'max-w-[400px]'
            }`}
```

File: web/src/components/modals/OnboardingModalInner.tsx [hunk 2]
```diff
@@ -116,14 +134,22 @@ function Modal({
...
                 />
               </picture>
             )}
             <h1>{currentView.title?.(t)}</h1>
           </div>
-          <div className="flex w-auto flex-col justify-center px-4 pt-6 text-center dark:bg-gray-700">
+          <div
+            className={`flex w-full flex-col justify-center px-4 pt-6 text-center ${
+              isMinHeight ? '' : 'pr-12'
+            }`}
```
Comment (VIKTORVAV99) [inline web/src/components/modals/OnboardingModalInner.tsx:151]:
```suggestion
            className={`flex w-full flex-col justify-center px-4 pt-6 text-center ${
              !isMinHeight && 'pr-12'
            }`}
```

**Review by VIKTORVAV99 — COMMENTED** [review]
Two small suggestions but otherwise it looks good!

Thanks for fixing the corners and centering!

File: web/src/components/modals/OnboardingModalInner.tsx (hunk 2 above)
Comment (noodlebenji2960) [reply web/src/components/modals/OnboardingModalInner.tsx:151]: nice and clean :+1:

**Pull request closed without merging**